### PR TITLE
Rerun all models with same training data/ test data split 

### PR DESCRIPTION
**Pull request by lizgzil**
- One random seed to rule them all
- add more to output file, 
- Script to run each models several times with different random seeds
- Experimentation notebook to look for best seed and variabilility in the results
- LogisticRegression(max_iter=1000) to solve an issue where it wasn't converging (although results seem the same anyway)
- Edit label_grants to evaluate the ensemble model as well

New ensemble results:
- This returned 4 models: ['count_SVM_201022', 'bert_SVM_scibert_201022', 'bert_SVM_bert_201022', 'tfidf_log_reg_201022']
- It found **1257** relevant grants.
- On the test data: **accuracy: 0.804
f1: 0.840
precision_score: 0.982
recall_score: 0.733**

| Number of models that need to agree | Number of relevant grants | Test F1 | Test precision | Test recall |
|---|---|---|---|---|
|1|5926| 0.919|0.860|0.987|
|2| 4125 |0.954 |0.936|0.973|
|3|2956|0.915|0.970|0.867|
|4|1257|0.840|0.982|0.733|

Previously:
- This returned 3 models: bert_log_reg_scibert_200807, bert_naive_bayes_scibert_200807, count_naive_bayes_200807
- It found 2163 grants
- f1:0.856, precision_score: 0.826, recall_score: 0.889 (this was calculated by averaging the model scores though so isn't comparable)

### CODE REVIEW

**Review by nonieA — APPROVED** [review]
This is great, and really good results aswell 👍

File: docs/Finding_Relevant_Grants.md [hunk 1]
```diff
@@ -257,7 +257,111 @@ test results:
...
+| actual tag 0|31|1|
+| actual tag 1|20|55|
+
+The above results are when all 4 models need to agree on a grant being relevant in order for the ensemble result to say it's relevant. However we can experiment with different numbers of the models needing to agree:
+
+| Number of models that need to agree | Number of relevant grants | Test F1 | Test precision | Test recall |
+|---|---|---|---|---|
+|2| 4125 |0.954 |0.936|0.973|
+|3|2956|0.915|0.970|0.867|
+|4|1257|0.840|0.982|0.733|
```
[the remark this answers is not in the document]
Comment (lizgzil) [reply docs/Finding_Relevant_Grants.md:367]: yes good point, i'll add that on monday and not close it until then 👍

File: nutrition_labels/grant_tagger.py [hunk 2]
```diff
@@ -7,8 +7,9 @@
 from sklearn.linear_model import LogisticRegression
 from sklearn.svm import SVC
 import pandas as pd
+import numpy as np
 
-from random import sample, seed
+from random import sample, seed, shuffle
```
[the remark this answers is not in the document]
Comment (lizgzil) [reply nutrition_labels/grant_tagger.py:12]: good spot, i dont think we do no

[the remark this answers is not in the document]
Comment (nonieA) [reply nutrition_labels/grant_tagger.py:12]: Its only because pycharm shows when they aren't referenced in the script

**Comment by lizgzil** [comment]
will merge now. Next action is to test fairness on both the 2 /4 and 3 /4 agreement ensemble models and then depending on those results decide which is best to use.
